pkg/machinery/config/types/v1alpha1/generate/example_test.go

```diff
@@ -0,0 +1,102 @@
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package generate_test
+
+import (
+	"log"
+	"os"
+
+	"github.com/talos-systems/talos/pkg/machinery/config"
+	v1alpha1 "github.com/talos-systems/talos/pkg/machinery/config/types/v1alpha1"
+	"github.com/talos-systems/talos/pkg/machinery/config/types/v1alpha1/generate"
+	"github.com/talos-systems/talos/pkg/machinery/config/types/v1alpha1/machine"
+	"github.com/talos-systems/talos/pkg/machinery/constants"
+)
+
+//nolint:wsl
+func Example() {
+	// This is an example of generating a set of machine configuration files for multiple
+	// nodes of the cluster from a single cluster-specific cluster.
+
+	// Input values for the config generation:
+
+	// * cluster name and Kubernetes control plane endpoint
+	clusterName := "test-cluster"
+	controlPlaneEndpoint := "https://kubernetes.example.com:6443"
+
+	// * Kubernetes version to install, using the latest here
+	kubernetesVersion := constants.DefaultKubernetesVersion
+
+	// * version contract defines the version of the Talos cluster configuration is generated for
+	//   generate package can generate machine configuration compatible with current and previous versions of Talos
+	targetVersion := "v1.0"
+
+	// parse the version contract
+	var (
+		versionContract = config.TalosVersionCurrent //nolint:wastedassign,ineffassign // version of the Talos machinery package
+		err             error
+	)
+
+	versionContract, err = config.ParseContractFromVersion(targetVersion)
+	if err != nil {
+		log.Fatalf("failed to parse version contract: %s", err)
+	}
+
+	// generate the cluster-wide secrets once and use it for every node machine configuration
+	// secrets can be stashed for future use by marshaling the structure to YAML or JSON
+	secrets, err := generate.NewSecretsBundle(generate.NewClock(), generate.WithVersionContract(versionContract))
+	if err != nil {
+		log.Fatalf("failed to generate secrets bundle: %s", err)
+	}
+
+	input, err := generate.NewInput(clusterName, controlPlaneEndpoint, kubernetesVersion, secrets,
+		generate.WithVersionContract(versionContract),
+		// there are many more generate options available which allow to tweak generated config programmatically
+	)
+	if err != nil {
+		log.Fatalf("failed to generate input: %s", err)
+	}
+
+	// generate the machine config for each node of the cluster using the secrets
+	for _, node := range []string{"machine1", "machine2"} {
+		var cfg *v1alpha1.Config
+
+		// generate the machine config for the node, using the right machine type:
+		// * machine.TypeConrolPlane for control plane nodes
+		// * machine.TypeWorker for worker nodes
+		cfg, err = generate.Config(machine.TypeControlPlane, input)
+		if err != nil {
+			log.Fatalf("failed to generate config for node %q: %s", node, err)
+		}
+
+		// config can be tweaked at this point to add machine-specific configuration, e.g.:
+		cfg.MachineConfig.MachineInstall.InstallDisk = "/dev/sdb"
```

Comment:
isn't it easier to use `generate.WithInstallDisk()`

Reply:
it is, but I wanted to give an example of config "patching". I first wanted to see if I should do our JSON patches, but the code looks so weird that I decided to do direct modification of the structure.